jocd/src/br/org/certi/jocd/dapaccess/Command.java

```diff
@@ -340,8 +399,12 @@ public void add(int count, byte request, long[] words, Byte dapIndex) {
    * Encode this command into a byte array that can be sent
    * The actual command this is encoded into depends on the data that was added.
    */
-  public byte[] encodeData() {
-    assert this.getEmpty() == false;
+  public byte[] encodeData() throws Error {
+    // Assert this.getEmpty() == false.
+    if (this.getEmpty()) {
+      throw new Error("encodeData: Unexpected getEmpty() value ( " + getEmpty() + ")");
```

Comment:
This getEmpty() call could be avoided here, since the error state already occurred and it's already known that the return value will be true.

Reply:
Fixed on 3905d3fb99c8c6b556bfe3b84b6ed06c2f55747c